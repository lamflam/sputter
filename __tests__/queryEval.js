const snapshot = require("../testUtil/snapshot");
const fixtures = require("../testUtil/fixtures");

describe(":eval", () => {
  snapshot(
    fixtures.smallProgram,
    ":eval('node.type === \"VariableDeclarator\"')"
  );

  snapshot(
    fixtures.smallProgram,
    // requires relative to process.cwd
    `:eval('node.type === require("./testUtil/variableDeclarator")')`
  );

  snapshot(
    fixtures.smallProgram,
    // require works with node modules
    `:eval('require("@babel/code-frame"), false')`
  );
});
