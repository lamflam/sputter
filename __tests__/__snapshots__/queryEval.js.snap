// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:eval :eval('node.type === "VariableDeclarator"') 1`] = `
> 1 | var x = 1;
    |     ^^^^^ VariableDeclarator
  2 | 
`;

exports[`:eval :eval('node.type === require("./testUtil/variableDeclarator")') 1`] = `
> 1 | var x = 1;
    |     ^^^^^ VariableDeclarator
  2 | 
`;

exports[`:eval :eval('require("@babel/code-frame"), false') 1`] = ``;
